### PR TITLE
server: use `async_broadcast` channel instead of delay in event loop

### PR DESCRIPTION
**Pull request by flouthoc**
We are using all most all of CPU with the approach where we are trying
to read value of kill switch in the event loop via tokio::time::Sleep. This approach looks bad.

Instead start listening to async broadcast channel as a kill switch
mechanism, this will make sure that we don't eat CPU for no reason.

Following also adds and uses a new crate: `async_broadcast` which is needed for mpmc ( multi-producer, multi-consumer ) broadcasting across threads.

Closes: https://github.com/containers/aardvark-dns/issues/46

### CODE REVIEW

**Comment by flouthoc** [comment]
@baude Please verify the patch and merge-me.

File: src/dns/coredns.rs [hunk 1]
```diff
@@ -18,14 +17,15 @@ use trust_dns_proto::{
...
-    backend: Arc<DNSBackend>,      // server's data store
-    kill_switch: Arc<Mutex<bool>>, // global kill_swtich
-    filter_search_domain: String,  // filter_search_domain
+    name: Name,                          // name or origin
+    address: IpAddr,                     // server address
+    port: u32,                           // server port
+    authority: InMemoryAuthority,        // server authority
+    cl: AsyncClient,                     //server client
+    backend: Arc<DNSBackend>,            // server's data store
+    kill_switch: Arc<Mutex<bool>>,       // global kill_swtich
```
Comment (giuseppe) [inline src/dns/coredns.rs:26]: typo in the comment

Comment (flouthoc) [reply src/dns/coredns.rs]: thanks fixed the typo.

**Comment by baude** [comment]
LGTM, appears to fix the problem

**Pull request merged**